docs/vm-installation.md

```diff
@@ -0,0 +1,144 @@
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/antrea-io/antrea/feature/externalnode/build/yamls/externalnode/vm-agent-rbac.yml
+```
+
+4. Create `antrea-agent.kubeconfig` file for `antrea-agent` to access the K8S API server.
+
+```bash
+export CLUSTER_NAME="kubernetes"
+export SERVICE_ACCOUNT="vm-agent"
+APISERVER=$(kubectl config view -o jsonpath="{.clusters[?(@.name==\"$CLUSTER_NAME\")].cluster.server}")
+TOKEN=$(kubectl -n vm-ns get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='$SERVICE_ACCOUNT')].data.token}"|base64 --decode)
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-cluster kubernetes --server=$APISERVER --insecure-skip-tls-verify=true
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-credentials antrea-agent --token=$TOKEN
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-context antrea-agent@kubernetes --cluster=kubernetes --user=antrea-agent
+kubectl config --kubeconfig=antrea-agent.kubeconfig use-context antrea-agent@kubernetes
+# Copy antrea-agent.kubeconfig to VM
+```
+
+5. Create `antrea-agent.antrea.kubeconfig` file for `antrea-agent` to access the `antrea-controller` API server.
+
+```bash
+# The VM should have network connectivity to reach the antrea-controller API server.
+export ANTREA_API_SERVER="https://172.18.0.1:443"
+# antrea-controller API server needs to be exposed via public IP or Node IP
+TOKEN=$(kubectl -n vm-ns get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='$SERVICE_ACCOUNT')].data.token}"|base64 --decode)
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-cluster antrea --server=$ANTREA_API_SERVER --insecure-skip-tls-verify=true
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-credentials antrea-agent --token=$TOKEN
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-context antrea-agent@antrea --cluster=antrea --user=antrea-agent
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig use-context antrea-agent@antrea
+# Copy antrea-agent.antrea.kubeconfig to VM
+```
+
+## Installation on Linux VM
+
+### Prerequisites
+
+OVS needs to be installed on the VM. For more information about OVS installation please refer to the [getting-started guide](getting-started.md#open-vswitch).
+
+### Installation
+
+1. Build `antrea-agent` binary in the root of the antrea code tree and copy the binary to Linux VM.
+
+```bash
+make docker-bin
+```
+
+2. Copy and update the `antrea-agent` configuration file which specifies agent configuration parameters. Edit
+   antrea-agent.conf file to set `clientConnection`,`antreaClientConnection` and `externalNodeNamespace` with the
+   correct values. Please refer to this [base configuration file](../build/yamls/externalnode/conf/antrea-agent.conf)
+   to check all the configuration parameters for VMs.
+
+```bash
+AGENT_NAMESPACE="vm-ns"
+KUBECONFIG_PATH="/var/run/antrea"
```

Comment:
How about adding commands to create the dirs and copy the files:

AGENT_BIN_PATH="/usr/sbin"
AGENT_CONF_PATH="/var/run/antrea"
mkdir -p AGENT_BIN_PATH AGENT_CONF_PATH
cp antrea-agent AGENT_BIN_PATH
cp antrea-agent.conf AGENT_CONF_PATH

Reply:
I have been going back and forth with this. IMO we can share example to modify configuration file.

When the user is configuring antrea-agent as a service, he can update the service configuration part according to his needs.

AGENT_BIN_PATH="/usr/sbin"
AGENT_CONF_PATH="/var/run/antrea"
AGENT_LOG_PATH="/var/log/antrea"
mkdir -p /var/log/antrea
cat << EOF > /etc/systemd/system/antrea-agent.service
Description="antrea-agent as a systemd service"
After=network.target
[Service]
ExecStart=$AGENT_BIN_PATH/antrea-agent \
--config=$AGENT_CONF_PATH/antrea-agent.conf \
--logtostderr=false \
--log_file=$AGENT_LOG_PATH/antrea-agent.log
Restart=on-failure
[Install]
WantedBy=multi-user.target
EOF